### ISSUE TITLE
Output repeated and deadlock with using the stweet.TweetsByIdRunner

### ISSUE THREAD

**Issue by soccerqi** [markowanga/stweet#82]
Hi @markowanga , when I use the TweetsByIdRunner function to scrape some tweets, it occasionally occurs a deadlock problem, that the reason is the outputs is repeated endlessly! 
For example, when tweet id is "1349479144225173509" will encounter this problem, my code is below:
```python
def try_tweet_by_id_scrap():
    id_task = st.TweetsByIdTask('1349479144225173509')
    filename = 'stweet_outputs/output_raw_id.{}.jl'.format(datetime.now().strftime('%Y-%m-%dT%H.%M.%S'))
    output_json = st.JsonLineFileRawOutput(filename)
    output_print = st.PrintRawOutput()
    st.TweetsByIdRunner(
        tweets_by_id_task=id_task,
        raw_data_outputs=[output_print, output_json],
        web_client=st.RequestsWebClient(proxy=proxy, interceptors=[TwitterAuthWebClientInterceptor()])
    ).run()


if __name__ == '__main__':
    try_tweet_by_id_scrap()
```

Thanks for help!

**Comment by soccerqi** [comment]
> Hi @markowanga , when I use the TweetsByIdRunner function to scrape some tweets, it occasionally occurs a deadlock problem, that the reason is the outputs is repeated endlessly! For example, when tweet id is "1349479144225173509" will encounter this problem, my code is below:
> 
> def try_tweet_by_id_scrap(): id_task = st.TweetsByIdTask('1349479144225173509') filename = 'stweet_outputs/output_raw_id.{}.jl'.format(datetime.now().strftime('%Y-%m-%dT%H.%M.%S')) output_json = st.JsonLineFileRawOutput(filename) output_print = st.PrintRawOutput() st.TweetsByIdRunner( tweets_by_id_task=id_task, raw_data_outputs=[output_print, output_json], web_client=st.RequestsWebClient(proxy=proxy, interceptors=[TwitterAuthWebClientInterceptor()]) ).run()
> 
> if **name** == '**main**': try_tweet_by_id_scrap()
> 
> Thanks for help!

My stweet version is 2.0.0, python version is 3.10.2, on macOS Monterey.

**Comment by markowanga** [comment]
Hello @soccerqi, I will focus on this. Next time please add complete code to reproduce error (with all imports and variables created) – I reproduced it for following code:
```python
from datetime import datetime

import stweet as st
from stweet.twitter_api.twitter_auth_web_client_interceptor import TwitterAuthWebClientInterceptor


def try_tweet_by_id_scrap():
    id_task = st.TweetsByIdTask('1349479144225173509')
    filename = 'output_raw_id.{}.jl'.format(datetime.now().strftime('%Y-%m-%dT%H.%M.%S'))
    output_json = st.JsonLineFileRawOutput(filename)
    output_print = st.PrintRawOutput()
    st.TweetsByIdRunner(
        tweets_by_id_task=id_task,
        raw_data_outputs=[output_print, output_json],
        web_client=st.RequestsWebClient(interceptors=[TwitterAuthWebClientInterceptor()])
    ).run()


if __name__ == '__main__':
    try_tweet_by_id_scrap()
```

**Comment by markowanga** [comment]
There was a small bug in ending condition -- I will fix it in 2.0.1
